dashboard/test/lib/projects_list_test.rb

```diff
@@ -264,8 +266,62 @@ class ProjectsListTest < ActionController::TestCase
     assert_equal combined_projects[:applab].last, {name: "recentApplab3"}
   end
 
+  test 'fetch_section_libraries filters by library_name' do
+    applab_lib_name = 'applab_library'
+    gamelab_lib_name = 'gamelab_library'
+    description = 'library description'
+    student_name = 'student name'
+    stub_projects = [
+      {
+        name: applab_lib_name,
+        properties: {}.to_json,
+        birthday: 13.years.ago.to_datetime,
+        storage_id: @storage_id,
+        id: 1,
+        project_type: 'applab',
+        value: {'libraryName': applab_lib_name, 'libraryDescription': description}.to_json,
+        state: 'active'
+      },
+      {
+        name: gamelab_lib_name,
+        properties: {}.to_json,
+        birthday: 13.years.ago.to_datetime,
+        storage_id: @storage_id,
+        id: 2,
+        project_type: 'gamelab',
+        value: {'libraryName': gamelab_lib_name, 'libraryDescription': description}.to_json,
+        state: 'active'
+      }
+    ]
+    stub_users = {
+      @storage_id => 4
+    }
+    Section = Struct.new(:students)
+    Student = Struct.new(:id, :name)
+    student = Student.new(4, student_name)
+    section = Section.new([student])
+
+    PEGASUS_DB.stubs(:[]).returns(user_db_result(stub_users)).then.returns(library_db_result(stub_projects))
+
+    StorageApps.stubs(:get_published_project_data).returns({})
+    lib_response = ProjectsList.send(:fetch_section_libraries, section)
+    assert_equal 2, lib_response.length
+    assert_equal applab_lib_name, lib_response[0][:name]
```

Comment:
Are the results guaranteed to come back in this order? if not, consider sorting `lib_response` by library name, or pulling out the library names and sorting those and comparing that array to the array of names you expect.

Reply:
Yeah, they'll come back in order. I don't know if they would if I used the DB, but since I'm mocking them, order should be preserved.

Reply:
oh right! :)